cabal-testsuite/src/Test/Cabal/Prelude.hs

```diff
@@ -966,7 +960,7 @@ hasNewBuildCompatBootCabal = isGhcVersion ">= 7.9"
 ------------------------------------------------------------------------
 -- * Broken tests
 
-expectBroken :: Int -> TestM a -> TestM ()
+expectBroken :: Int -> TestM a -> TestM a
```

Comment:
There is something weird with the expecting broken tests I think. If I have a test that succeeds but the expected output doesn't match, then it is reported as fail. If I mark it as expectedBroken, then it says that the test was an unexpected succeed. My suspicion is that the step for checking the output is not considered when dealing with whether a test is broken or not.

This results in "test broken by wrong output" to necessarily be skipped and not marked as broken.

Reply:
That analysis sounds right. Probably the function which does this is `requireSuccess`.